streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,68 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = segments.allSegments();
+        
+        final Segment minSegment = segments.minSegment();
+        final Segment maxSegment = segments.maxSegment();
+        final Bytes minKey = minSegment.first().key;
+        final Bytes maxKey = maxSegment.last().key;
+        
+        final long minTimestamp = keySchema.segmentTimestamp(minKey);
+        final long maxTimestamp = keySchema.segmentTimestamp(maxKey);
+        final Bytes keyFrom = WindowStoreUtils.bytesKeyFromBinaryKey(minKey.get());
+        final Bytes keyTo = WindowStoreUtils.bytesKeyFromBinaryKey(maxKey.get());
+        
+        return new SegmentIterator(searchSpace.iterator(),
+                                   keySchema.hasNextCondition(keyFrom, keyTo, minTimestamp, maxTimestamp),
+                                   minKey, maxKey);
+    }
+    
+    @Override
+    public KeyValueIterator<Bytes, byte[]> fetchAll(final long timeFrom, final long timeTo) {
```

Comment:
I'm wondering if this method could just be:
```
final Bytes key = new Bytes(new byte[0]);
return fetch(key, key, timeFrom, timeTo +1);
```
@guozhangwang ?

Reply:
Test for this method?

Reply:
I ran your above suggestion through my tests, and it didn't pass. I think the substitute you used for keyTo was too small.

Reply:
Similarly to what I mentioned above, we can use the extended `SegmentIterator` again with the `segments` parameter passed from the existing `List<Segment> Segments#segments`. By doing this we can also get rid of the `getMinSegmentGreaterThanEqualToTimestamp` and `getMaxSegmentLessThanEqualToTimestamp` as @dguy suggested.